pennylane_qiskit/converter.py

```diff
@@ -36,17 +36,83 @@
+    """Processes the inputs for calling the quantum function and returns
+    a dictionary of the format ``{Parameter("name"): value}`` for all the parameters.
+
+    For a ``quantum_circuit`` with parameters ``[Parameter("phi"), Parameter("psi"), Parameter("theta")]``,
+    inputs can be one of the following:
+        1. the kwargs passed to when calling the qfunc, other than ``params`` and ``wires``.
+            The keys in kwargs are expected to correspond to the names of the Parameters on
+            the QuantumCircuit, i.e. ``(qc, None, phi=0.35, theta=0.2, psi=1.7)``
+        2. the args passed when calling the qfunc, i.e. ``(qc, None, 0.35, 1.7, 0.2)``. These
+           are assigned to the Parameters as the are ordered on the Qiskit QuantumCircuit:
+           alphabetized by Parameter name
+        3. Some combination of args and kwargs, i.e. ``(qc, None, 0.35, 0.2, psi=1.7)``
+        4. (legacy) ``params`` from the kwarg ``params`` of the qfunc call, which is expected
+            to already be a dictionary of the format ``{Parameter("name"): value}``, i.e.
+            ``(qc, {Parameter("phi"): 0.35, Parameter("psi"): 1.7, Parameter("theta"): 0.2})``
+        5. (legacy) ``params`` passed as a single arg, which is expected
+            to already be a dictionary of the format ``{Parameter("name"): value}``, i.e.
+            ``(qc, None, {Parameter("phi"): 0.35, Parameter("psi"): 1.7, Parameter("theta"): 0.2})``
+
+    Returns:
+        params (dict): A dictionary mapping ``quantum_circuit.parameters`` to values
+    """
+
+    # if no kwargs are passed, and a dictionary has been passed as a single argument, then assume it is params
+    if params is None and not kwargs and (len(args) == 1 and isinstance(args[0], dict)):
+        return args[0]
+
+    # make params dict if using args and/or kwargs
+    if args or kwargs:
+        if params is not None:
+            raise RuntimeError(
+                "Cannot define params via the params kwarg when passing Parameter values "
+                "as individual args or kwargs."
+            )
+
+        # create en empty params dict
+        params = {}
+
+        # populate it with any parameters defined as kwargs
+        for k, v in kwargs.items():
+            # the key needs to be the actual Parameter, whereas kwargs keys are parameter names
+            qc_param = [p for p in quantum_circuit.parameters if p.name == k]
+            if not qc_param:
+                param_names = ", ".join([p.name for p in quantum_circuit.parameters])
+                raise TypeError(
+                    f"Got unexpected parameter keyword argument '{k}'. Circuit contains parameters: {param_names}"
+                )
+            params[qc_param[0]] = v
+
+        # get any parameters not defined in kwargs (may be all of them) and match to args in order
+        arg_parameters = [p for p in quantum_circuit.parameters if p.name not in kwargs.keys()]
```

Comment:
```suggestion
        arg_parameters = [p for p in quantum_circuit.parameters if p.name not in kwargs]
```

something pylint is locally complaining about.